### ISSUE TITLE
Markers not being removed from MarkerClusterer

### ISSUE THREAD

**Issue by 11th** [tomchentw/react-google-maps#153]
Consider this example:

The Sydney marker should just stay in place, but after every even click, the cluster number increments.
I suspect that removing is not handled properly...

```cjsx
React               = require 'react'
{GoogleMap, Marker} = require 'react-google-maps'
MarkerClusterer     = require 'react-google-maps/lib/addons/MarkerClusterer'

Sydney =
	lat: -33.8539657
	lng: 150.9485251

ClusterDemo = React.createClass
	
	getInitialState: ->
		inside: false

	handleToggle: ->
		@setState
			inside: !@state.inside

	render: ->
		<div>
			<div>
				<Button onClick={@handleToggle}>Toggle</Button>
			</div>
			<div style={height: '200px', width: '600px'}>
				<GoogleMap
						containerProps={{style: {height: '100%'}}}
						defaultZoom={3}
						defaultCenter={Sydney}
					>
					<MarkerClusterer ref="cluster">
						{if @state.inside
							<Marker position={Sydney} />
						}
					</MarkerClusterer>

					{if !@state.inside
						<Marker position={Sydney} />
					}
				</GoogleMap>
			</div>
		</div>

module.exports = ClusterDemo
```

**Comment by 11th** [comment]
Why there are no tests for simple cases as this? @tomchentw

**Comment by tomchentw** [comment]
@11th welcome for some test cases!

**Comment by tomchentw** [comment]
Released [v4.4.1](https://github.com/tomchentw/react-google-maps/releases/tag/v4.4.1)

**Comment by tomchentw** [comment]
Ref #146
